domain/sysvar_cache.go

```diff
@@ -229,7 +229,14 @@ func (do *Domain) checkEnableServerGlobalVar(name, sVal string) {
 		}
 		topsqlstate.GlobalState.ReportIntervalSeconds.Store(val)
 	case variable.TiDBRestrictedReadOnly:
-		variable.RestrictedReadOnly.Store(variable.TiDBOptOn(sVal))
+		on := variable.TiDBOptOn(sVal)
+		if on {
+			// when restricted readonly is on, super readonly must be on
+			variable.VarTiDBSuperReadOnly.Store(on)
+		}
+		variable.RestrictedReadOnly.Store(on)
+	case variable.TiDBSuperReadOnly:
+		variable.VarTiDBSuperReadOnly.Store(variable.TiDBOptOn(sVal))
```

Comment:
Can you make the code here match the `SetGlobal` func for the sysvar?

We only execute this code here when receiving a notification from etcd, but the behavior should be the same when running without etcd so we can test in the testsuite.

(In future, we might eliminate the code here and run the `SetGlobal` func on each tidb-instance, but that's not the case currently.)

Reply:
sure